File: capacity-club-api/src/domain-modules/indoor-training-subscription/model/payload/indoor-training-subscription-create.payload.ts
import { Payload } from '@domain-modules-shared';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsOptional } from 'class-validator';
import { IndoorTraining } from 'domain-modules/indoor-training/model';
import { Member } from 'domain-modules/member/model';

export class IndoorTrainingSubscriptionCreatePayload implements Payload {
  @ApiProperty()
  @IsNotEmpty()
  purchase_date: Date;
  @ApiProperty({ type: () => IndoorTraining })
  @IsOptional()
  indoor_training: IndoorTraining;
  @ApiProperty({ type: () => Member })
  @IsOptional()
  member: Member;
}

File: capacity_club_mobile_app/lib/domain-features/data/indoor-training-subscription/model/payload/indoor_training_subscription_create_payload.dart
import 'package:capacity_club_mobile_app/core/model/abstract/payload.dart';
import 'package:capacity_club_mobile_app/domain-features/data/indoor-training/model/indoor_training_model.dart';
import 'package:capacity_club_mobile_app/domain-features/data/member/model/member_model.dart';
import 'package:equatable/equatable.dart';

class IndoorTrainingSubscriptionCreatePayload extends Equatable
    implements Payload {
  final DateTime purchased_date;
  final MemberModel? member;
  final IndoorTrainingModel? indoor_training;

  IndoorTrainingSubscriptionCreatePayload({
    required this.purchased_date,
    this.member,
    required this.indoor_training,
  });

  Map<String, dynamic> toJson() {
    return {
      'purchased_date': purchased_date.toIso8601String(),
      'member': member ?? {},
      'indoor_training': indoor_training ?? {}
    };
  }

  @override
  List<Object?> get props => [
        purchased_date,
        member,
        indoor_training,
      ];
}
